src/cmd/top/top.c

```diff
@@ -237,8 +238,49 @@ static char * build_title (struct top *top, const char *title)
     return strdup (title);
 }
 
+static void setup_constraint (struct top *top)
+{
+    flux_future_t *f;
+    json_t *o;
+    json_t *tmp;
+    json_t *requires = NULL;
+
+    if (!(f = flux_rpc (top->h, "config.get", NULL, FLUX_NODEID_ANY, 0))
+        || flux_rpc_get_unpack (f, "o", &o) < 0)
+        fatal (errno, "Error fetching flux config");
+
```

Comment:
Suggestion: fetch the config object and store it within `struct top` before calling `setup_constraint()` ?   Seems like it might come in handy later e.g. when multiple queues/constraints are supported.